### PR TITLE
[web] Ensure handled key event is not propagated to IME

### PR DESCRIPTION
**Pull request by knopp**
Fixes [136460](https://github.com/flutter/flutter/issues/136460)

Changes:
 - Raw keyboard event is handled during capture phase. This is to ensure that the framework processes the event before reaching to IME text area and raw keyboard can stop the propagation for handled events.
 - `RawKeyboard` event handler is invoked from `KeyboardBinding` event handler. This is to prevent race condition because both handlers now run in capture phase and `KeyboardBinding` needs to process the event first.

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by dkwingsmt** [comment]
The change looks ok, but it doesn't seem to me that the unit test this PR modifies is related to the change. Is this change tested?

**Comment by knopp** [comment]
> The change looks ok, but it doesn't seem to me that the unit test this PR modifies is related to the change. Is this change tested?

The test was missing indeed. Added now, can you take another look?

File: third_party/web_locale_keymap/lib/web_locale_keymap/locale_keymap.dart [hunk 1]
```diff
@@ -41,6 +41,9 @@ class LocaleKeymap {
       return eventKeyCode;
     }
     if (result == null) {
+      if ((eventCode ?? '').isEmpty && (eventKey ?? '').isEmpty) {
```
Comment (Renzo-Olivares) [inline third_party/web_locale_keymap/lib/web_locale_keymap/locale_keymap.dart:44]: nit: could this be `eventCode == null && eventKey == null` ?

**Review by Renzo-Olivares — APPROVED** [review]
LGTM with a small nit. Thanks for the change!

File: third_party/web_locale_keymap/lib/web_locale_keymap/locale_keymap.dart (hunk 1 above)
Comment (knopp) [reply third_party/web_locale_keymap/lib/web_locale_keymap/locale_keymap.dart:44]: No. We're passing empty strings there during some tests unfortunately.

**Comment by knopp** [comment]
@dkwingsmt, friendly ping

**Comment by dkwingsmt** [comment]
Is anything related to `stopPropagation` tested? Actually I thought this was the major change of this PR.

**Review by dkwingsmt — APPROVED** [review]

**Pull request merged**